### PR TITLE
Fix BottomSheet focus issues

### LINKED ISSUE #563: Focus when expanding and collapsing BottomSheet with VoiceOver and TalkBack
**Scenario** to reproduce
When the sheet is expanded the focus follow the "COVID alert is active and jumps back to the bottom of the page view. 
When the sheet is collapsed the focus has inconsistent positioning. It should orient people to the header or. 

**Expected behaviour**
When BottomSheet is expanded, it should focus on first card header.
When BottomSheet is collapsed, it should focus on header in the content. 

**Bug or issue / WCAG**
2.4.3 Focus Order

**Video of experience**
https://drive.google.com/drive/u/0/recent

I am combining these issues here: 
https://github.com/cds-snc/covid-shield-mobile/issues/544
https://github.com/cds-snc/covid-shield-mobile/issues/542
https://github.com/cds-snc/covid-shield-mobile/issues/488

### PR DESCRIPTION
**Pull request by nickmly**
Fixes #563 

Tested with Talkback on Android. Screen reader should now:

- [x] Focus on and read title of HomeScreen view when BottomSheet is closed
- [x] Focus on StatusHeaderView when opening the BottomSheet
- [x] Scroll through BottomSheet

### CODE REVIEW

File: src/screens/home/views/OverlayView.tsx [hunk 1]
```diff
@@ -121,6 +122,22 @@ const ShareDiagnosisCode = ({i18n}: {i18n: I18n}) => {
...
 
+const AccessibleView = ({children}: {children: React.ReactNode}) => {
+  const [screenReaderEnabled, setScreenReaderEnabled] = useState(false);
+  const handleScreenReaderToggled = (screenReaderEnabled: any) => {
+    setScreenReaderEnabled(screenReaderEnabled);
+  };
+  AccessibilityInfo.addEventListener('screenReaderChanged', handleScreenReaderToggled);
+
+  return screenReaderEnabled ? (
+    <ScrollView style={styles.content} showsVerticalScrollIndicator={false}>
```
Comment (henrytao-me) [inline src/screens/home/views/OverlayView.tsx:133]: Nice work 👍

File: src/screens/home/views/OverlayView.tsx [hunk 2]
```diff
@@ -121,6 +122,22 @@ const ShareDiagnosisCode = ({i18n}: {i18n: I18n}) => {
   );
 };
 
+const AccessibleView = ({children}: {children: React.ReactNode}) => {
+  const [screenReaderEnabled, setScreenReaderEnabled] = useState(false);
+  const handleScreenReaderToggled = (screenReaderEnabled: any) => {
+    setScreenReaderEnabled(screenReaderEnabled);
+  };
+  AccessibilityInfo.addEventListener('screenReaderChanged', handleScreenReaderToggled);
```
Comment (henrytao-me) [inline src/screens/home/views/OverlayView.tsx:130]: We need to use `useEffect` and clean up listener proper. I suggest to create AccessibilityProvider at root (App.tsx) and provide a hook to get `screenReaderEnabled`.

File: src/components/Text.tsx [hunk 3]
```diff
@@ -8,12 +8,13 @@ import {useAccessibilityAutoFocus} from 'shared/useAccessibilityAutoFocus';
 export type TextProps = RestyleTextProps<Theme> &
   Omit<React.ComponentProps<typeof RNText> & {children?: React.ReactNode}, keyof RestyleTextProps<Theme>> & {
     accessibilityAutoFocus?: boolean;
+    focusRef?: React.Dispatch<any>;
```
Comment (henrytao-me) [inline src/components/Text.tsx:11]: `React.Dispatch` is not the right type for ref. It should be `React.LegacyRef`

**Comment by nickmly** [comment]
> Let's optimize `useAccessibilityAutoFocus` hook too. If we have AccessibilityProvider, we don't need to call `AccessibilityInfo.isScreenReaderEnabled()`.

This has been done, thank you @henrytao-me!

**Pull request merged**